### DISCUSSION TITLE
Can the dataset of graph classification task be used for node classification training?

### DISCUSSION THREAD

**Discussion by jiaruHithub** [pyg-team/pytorch_geometric#2650] [Q&A] [answered]
Dear developers:
I want to know: Can the dataset of graph classification task be used for node classification training?
Such as ```PROTEINS``` and ```NCI1```.
If this is possible, where are the examples?
Thank you for your answer!

**Comment by rusty1s** [comment, marked as answer]
A graph classification dataset contains only a single label per graph, while in node classification you have a label for each node. If you want to cast a graph classification dataset to a node classification one, I think it's reasonable to set each node label to the ground-truth label of the given class, *e.g.*:
```python
def transform(data):
    data.y = data.y.repeat(data.num_nodes)
    return data

dataset = TUDataset(..., transform=transform)
```

**Reply by jiaruHithub** [reply]
Thank you!
